frontend/src/reducers/user.js

```diff
@@ -0,0 +1,23 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const user = createSlice ({
+    name: 'user', 
+    initialState: {
+        username: null,
+        accessToken: null,
+        errors: null
+    },
+    reducers: {
+        setUsername: (store, action) => {
+            store.username = action.payload
+        },
+        setAccessToken: (store, action) => {
+            store.accessToken = action.payload
+        },
+        setErrors: (store, action) => {
+            store.errors = action.payload
+        }
+    }
```

Comment:
You created a nice reducer here, but you forgot to save your user to the localStorage to prevent logging out during refreshing.